include/envoy/common/platform.h

```diff
@@ -151,6 +152,11 @@ struct msghdr {
 #define HANDLE_ERROR_PERM ERROR_ACCESS_DENIED
 #define HANDLE_ERROR_INVALID ERROR_INVALID_HANDLE
 
+#define ENVOY_SIGINIT CTRL_C_EVENT
+#define ENVOY_SIGTERM CTRL_CLOSE_EVENT
+#define ENVOY_SIGUSR1 -1
+#define ENVOY_SIGHUP -1
```

Comment:
It seems that CTRL_LOGOFF_EVENT and CTRL_SHUTDOWN_EVENT must follow the ENVOY_SIGTERM path as any linux would cause during shutdown, or envoy will have no chance to exit cleanly but os shutdown will be delayed.

The CTRL_BREAK_EVENT is really interesting; this could be mapped to graceful shutdown or hot restart.

Reply:
(Also CTRL SHUTDOWN is quite relevant for running Envoy via Windows containers as I believe that is the only notification/event a process will be able to handle when a container is being destroyed. Container orchestrators may also choose to do some lifecycle management with the Admin API `/quitquitquit` endpoint instead, but handling CTRL SHUTDOWN would cover other cases.)

Reply:
Is your suggestion to register some additional signal events so they also go down the clean shutdown path?  Perhaps we should add a function that returns a list of signals that should trigger the clean shutdown handler

Should config for fast vs clean shutdown on CTRL_LOGOFF_EVENT and CTRL_BREAK_EVENT make sense?  I could see fast teardown being desirable on logoff.  Clean shutdown on both of these events may be most appropriate.

Reply:
All the signals in the control handler map to clean shutdown so I just created an event `ENVOY_WIN32_SIGTERM` and all the `CtrlHandler` events are mapped to it. 

All the Windows events are causing a clean shutdown so we dont need a function for now. When we add support for SCM if might have other events that are not related to shutdown and at that point we might need to implement a function

Reply:
Is there a concept of clean vs fast shutdown?